terminusdb_client/client/Client.py

```diff
@@ -1481,16 +1481,30 @@ def has_doc(self, doc_id: str, graph_type: str = "instance") -> bool:
         graph_type : str
             Graph type, either "instance" or "schema".
 
+        Raises
+        ------
+        InterfaceError
+            if the client does not connect to a database
+
         Returns
         -------
         Bool
             if the document exist
         """
         self._validate_graph_type(graph_type)
         self._check_connection()
-        all_existing_obj = self.get_all_documents(graph_type=graph_type)
-        all_existing_id = [x.get("@id") for x in all_existing_obj]
-        return doc_id in all_existing_id
+
+        response = requests.get(
+            self._documents_url(),
+            headers=self._default_headers,
+            json={"id": doc_id, "graph_type": graph_type},
+            auth=self._auth(),
+        )
+        try:
+            _finish_response(response)
+            return True
+        except DatabaseError:
+            return False
```

Comment:
Should you not be checking what sort of database error this is? It could be completely unrelated to the nonexistence of a document.

Reply:
That is so true